main.go

```diff
@@ -15,6 +15,138 @@
 
 package main
 
+import (
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cast"
+
+	"github.com/bank-vaults/secret-init/logger"
+	"github.com/bank-vaults/secret-init/providers"
+	"github.com/bank-vaults/secret-init/providers/vault"
+)
+
+var providersMap = map[string]providers.Provider{
+	// "aws": aws.NewAWSProvider(),
+	// "gcp": gcp.NewGCPProvider(),
+	"vault": vault.NewVaultProvider(),
+}
+
 func main() {
-	println("hello world")
+	logger := logger.SetupSlog()
+
+	var providerName string
+	if len(os.Args) >= 3 && os.Args[1] == "-p" {
+		providerName = os.Args[2]
+
+		// Remove the "-p {provider name}" argument from the command-line arguments
+		os.Args = append(os.Args[:1], os.Args[3:]...)
```

Comment:
Big no-no: store the args in a separate variable if necessary, but do not modify global variables.